### PR TITLE
MAGECLOUD-4322: Automate test for environment variables validation

### PR DESCRIPTION
**Pull request by andriyShevtsov**
### Description
This test is automates  testing of data types validation for environment variables

### Fixed Issues (if relevant)

1. Jira task https://magento2.atlassian.net/browse/MAGECLOUD-4322 
2. ...

### Manual testing scenarios

1. There no Zephyr test which is exactly the same to this but [this one](https://jira.corp.magento.com/browse/MAGECLOUD-47) is the most closer

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
+<?php
```
Comment (oshmyheliuk) [inline src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php:1]: > Automate test for .magento.env.yaml validation

I don't see any validation of  .magento.env.yaml file in this test. Looks like this test for validation variables from UI

File: src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    /**
+     * @param \CliTester $I
+     * @param \Codeception\Example $data
+     * @throws \Robo\Exception\TaskException
+     * @dataProvider dataTypesDataProvider
+     */
+    public function dataTypesValidationOnDeploy(\CliTester $I, \Codeception\Example $data)
+    {
+      //  $I->assertTrue($I->uploadToContainer($data['app_yaml'], '/.magento.app.yaml', Docker::BUILD_CONTAINER));
```
Comment (oshmyheliuk) [inline src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php]: Please remove the commented code.

File: src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php (hunk 1 above)
Comment (andriyShevtsov) [reply src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php:1]: You are right. It's typo

File: src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php (hunk 2 above)
Comment (andriyShevtsov) [reply src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php]: Done

File: src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
```
Comment (oshmyheliuk) [inline src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php:6]: add an empty line after declare

File: src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    protected function dataTypesDataProvider(): array
+    {
+        return [
+            'string_instead_integer' => [
+                'cloudVariables' => [
+                    'MAGENTO_CLOUD_VARIABLES' => [
+                        'SCD_THREADS' => 'one',
+                    ],
+                ],
+                'expectedError' => 'wrong value',
```
Comment (oshmyheliuk) [inline src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php]: Please add more specific expected Error message, `wrong value` can match an error for a different variables.

File: src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            ],
+            'integer_instead_boolean' => [
+                'cloudVariables' => [
+                    'MAGENTO_CLOUD_VARIABLES' => [
+                        'STATIC_CONTENT_SYMLINK' => 1,
+                    ],
+                ],
+                'expectedError' => 'wrong value',
```
Comment (oshmyheliuk) [inline src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php]: The same as above

File: src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php (hunk 3 above)
Comment (andriyShevtsov) [reply src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php:6]: done

File: src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php (hunk 4 above)
Comment (andriyShevtsov) [reply src/Test/Functional/Acceptance/DataTypesOptionValidationCest.php]: done

**Review by oshmyheliuk — APPROVED** [review]

**Review by shiftedreality — APPROVED** [review]

**Comment by shiftedreality** [comment]
Awaiting for Travis Ci to be fixed

**Pull request merged**